### ISSUE TITLE
Disable Dummy on Solo Maps

### ISSUE THREAD

**Issue by teini94** [ddnet/ddnet#5697]
add it to sv_solo_server 1 to prevent multiple finishes 
we cant just players_per_ip 1 cause of timeout

**Comment by VoxelDoesCode** [comment]
What's wrong with having both you and your dummy play the map at the same time?

**Comment by C0D3D3V** [comment]
`players_per_ip 1` is no solution, since that would prevent that two persons in the same room playing the same solo map. 
Also I do not see a problem with having the dummy finish at the same time, because you used all the time dummy copy.

**Comment by def-** [comment]
It's annoying because it moves everyone else one rank down further. But we can't prevent it easily.

**Comment by teini94** [comment]
we could even disable connecting dummys in ictf server

**Comment by JSaurusRex** [comment]
Server flags are send to the player when they connect if I remember correctly, so you can disable the dummy flag when the map reloads. There is probably a way to still get around it since it's purely a UI change as far as I know but it helps a lot of cases since most people won't go through the effort.

**Comment by fokkonaut** [comment]
https://github.com/ddnet/ddnet/pull/4358

**Comment by Chairn** [comment]
It also gives an "unfair" advantage where someone can try faily part twice by disabling dummy_copy just before the hard part.

**Comment by C0D3D3V** [comment]
I guess for testing hard parts you would just put your dummy in a perma practice team and tp it any time to you... You do not need dummy_copy. Thats what player actually do on faily maps, with very faily parts. I have seen that multiple times. Since this is in theory also possible with an second client, I do not see this as a problem. I only see the problem def- has mentioned.

**Comment by Chairn** [comment]
After you trained the part, you still use dummy_copy to have 2 chances of trying a legit run, so it still gives unfair advantage.

**Comment by C0D3D3V** [comment]
True, I think allowing to set the SERVERCAPFLAG_ALLOWDUMMY via map setting or generally for solo maps, would be ok.

**Comment by heinrich5991** [comment]
`SERVERCAPFLAG_ALLOWDUMMY` is a general flag for the server, it can't be changed on map change. We'd have to come up with a different message for it.

**Comment by fokkonaut** [comment]
I had it implemented as a GameInfoFlag first, but your opinion was it belongs to the Servercap flags :D
What a pity :/

**Comment by AssassinTee** [comment]
The current implementation only allows/disallows dummies for all of the server. It would be nice if we could expand it with a config variable for the map. A lot of solo maps already have `sv_solo_server 1` backed into them. Next apparently `copymoves` is abused on older solo maps, which could also be prevented by adding a config variable for this
